agent/exec/container/controller.go

```diff
@@ -194,6 +195,12 @@ func (r *controller) Wait(pctx context.Context) error {
 				// If we get here, something has gone wrong but we want to exit
 				// and report anyways.
 				return ErrContainerDestroyed
+			case "health_status: unhealthy":
```

Comment:
We should use event. Can we add an integration test to make sure the event is triggered.